plan/logical_plan_builder.go

```diff
@@ -612,6 +612,14 @@ func (b *planBuilder) buildDistinct(child LogicalPlan, length int) LogicalPlan {
 // joinFieldType finds the type which can carry the given types.
 func joinFieldType(a, b *types.FieldType) *types.FieldType {
 	resultTp := types.NewFieldType(types.MergeFieldType(a.Tp, b.Tp))
+	// This logic need be combined with buildProjection4Union.
```

Comment:
I think we'd better not embed this function to `buildProjection4Union`. this function is focused on deciding the data type for each column of the `union`.

Reply:
This logic is intelligible when it is associated with the buildProjection4Union logic.